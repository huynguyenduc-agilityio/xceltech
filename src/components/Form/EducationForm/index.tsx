import { useForm, useWatch } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';

// Constants
import { MESSAGES } from '@/constants';

// Types
import {
  EducationType,
  IEmployeeEducationInfo,
  MutationType,
  ToastStatus,
  EducationFormValues,
} from '@/types';

// Hooks
import { useEducationMutation, useToast } from '@/hooks';

// Utils
import { educationSchema, formatDate } from '@/utils';

// Components
import {
  Button,
  DatePicker,
  Form,
  FormControl,
  FormField,
  FormItem,
  FormMessage,
  Label,
  Textarea,
  TextField,
} from '@/components';

interface IEducationForm {
  mode: MutationType;
  section: EducationType;
  initialValues?: Partial<IEmployeeEducationInfo>;
  onBack?: () => void;
}

const EducationForm = ({
  initialValues = {},
  mode,
  section,
  onBack,
}: IEducationForm) => {
  const { toast } = useToast();
  const { handleEducationMutation, isEducationMutationLoading } =
    useEducationMutation({
      type: mode,
    });

  const {
    name = '',
    course = '',
    department = '',
    description = '',
    startDate,
    endDate,
    location = '',
  } = initialValues;

  const defaultValues: IEmployeeEducationInfo = {
    name,
    course,
    department,
    description,
    endDate,
    location,
    startDate,
  };

  const form = useForm<EducationFormValues>({
    mode: 'onBlur',
    reValidateMode: 'onChange',
    resolver: zodResolver(educationSchema),
    defaultValues,
  });

  const {
    control,
    handleSubmit,
    reset,
    getValues,
    formState: { isValid, isDirty, isSubmitting },
  } = form;

  const isCreate = mode === MutationType.Create;

  const onSubmit = async (data: IEmployeeEducationInfo) => {
    const payload = {
      ...data,
      type: section,
      startDate: formatDate(data.startDate || ''),
      endDate: formatDate(data.endDate || ''),
      ...(!isCreate && initialValues?.id ? { id: initialValues.id } : {}),
    };

    try {
      await handleEducationMutation(payload);

      toast({
        status: ToastStatus.Success,
        title: isCreate
          ? MESSAGES.COMMON.ADD_SUCCESS('Education')
          : MESSAGES.COMMON.UPDATE_SUCCESS('Education'),
      });
      onBack?.();

      reset(initialValues);
    } catch {
      toast({
        status: ToastStatus.Error,
        title: isCreate
          ? MESSAGES.COMMON.ADD_FAILED('Education')
          : MESSAGES.COMMON.UPDATE_FAILED('Education'),
      });

      reset(getValues());
    }
  };

  const disableSubmit = !isDirty || !isValid || isSubmitting;
  const startDateValue = useWatch({ control, name: 'startDate' }) as Date;
  const endDateValue = useWatch({ control, name: 'endDate' }) as Date;

  return (
    <Form {...form}>
      <form onSubmit={handleSubmit(onSubmit)}>
        <div className="w-full text-black-default space-y-10">
          <div className="space-y-8 mt-12">
            <div className="flex items-start gap-9">
              <FormField
                control={control}
                name="name"
                render={({ field, fieldState: { error } }) => (
                  <TextField
                    label="Name of Institution"
                    variant="primary"
                    className="space-y-5"
                    labelClassName="text-md"
                    inputClassName="h-[68px] px-12 rounded-[15px]"
                    placeholder="Enter your institution name"
                    {...field}
                    errorMessage={error?.message}
                  />
                )}
              />
              <FormField
                control={control}
                name="department"
                render={({ field, fieldState: { error } }) => (
                  <TextField
                    label="Department"
                    variant="primary"
                    className="space-y-5"
                    labelClassName="text-md"
                    inputClassName="h-[68px] px-12 rounded-[15px]"
                    placeholder="Enter your department"
                    {...field}
                    errorMessage={error?.message}
                  />
                )}
              />
            </div>
            <div className="flex items-start gap-9">
              <FormField
                control={control}
                name="course"
                render={({ field, fieldState: { error } }) => (
                  <TextField
                    label="Course"
                    variant="primary"
                    className="space-y-5"
                    labelClassName="text-md"
                    inputClassName="h-[68px] px-12 rounded-[15px]"
                    placeholder="Enter your course"
                    {...field}
                    errorMessage={error?.message}
                  />
                )}
              />
              <FormField
                control={control}
                name="location"
                render={({ field, fieldState: { error } }) => (
                  <TextField
                    label="Location"
                    variant="primary"
                    className="space-y-5"
                    labelClassName="text-md"
                    inputClassName="h-[68px] px-12 rounded-[15px]"
                    placeholder="Enter your location"
                    {...field}
                    errorMessage={error?.message}
                  />
                )}
              />
            </div>
            <div className="flex items-start gap-9">
              <FormField
                control={control}
                name="startDate"
                render={({ field: { value, onChange } }) => (
                  <FormItem className="space-y-5">
                    <Label className="flex flex-1 text-md">Start Date</Label>
                    <FormControl>
                      <DatePicker
                        date={value as Date}
                        disabledRange={(date) => date > endDateValue}
                        placeholder="Enter your start date"
                        onSelect={onChange}
                        className="px-11"
                      />
                    </FormControl>
                  </FormItem>
                )}
              />
              <FormField
                control={control}
                name="endDate"
                render={({ field: { value, onChange } }) => (
                  <FormItem className="space-y-5">
                    <Label className="flex flex-1 text-md">End Date</Label>
                    <FormControl>
                      <DatePicker
                        date={value as Date}
                        disabledRange={(date) => date < startDateValue}
                        placeholder="Enter your end date"
                        onSelect={onChange}
                        className="px-11"
                      />
                    </FormControl>
                  </FormItem>
                )}
              />
            </div>
          </div>
          <FormField
            key={name}
            control={control}
            name="description"
            render={({ field }) => (
              <FormItem className="space-y-5">
                <Label className="text-md">Description</Label>
                <FormControl>
                  <Textarea
                    {...field}
                    className="h-[122px] px-9 rounded-[15px]"
                    placeholder="Enter your description"
                  />
                </FormControl>
                <FormMessage />
              </FormItem>
            )}
          />

          <div className="w-full h-[70px]">
            <Button
              type="submit"
              className="w-[364px] bg-green-primary"
              isLoading={isEducationMutationLoading}
              disabled={disableSubmit}
            >
              {isCreate ? 'Submit' : 'Update'}
            </Button>
          </div>
        </div>
      </form>
    </Form>
  );
};

export default EducationForm;
