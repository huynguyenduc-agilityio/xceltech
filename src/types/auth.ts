export interface ILoginUser {
  email: string;
  password: string;
}

export interface IAuthUser {
  refresh: string;
  access: string;
  user: {
    id: string;
    email: string;
  };
}

export interface IRegisterUser extends ILoginUser {
  firstName: string;
  lastName: string;
  phone: string;
  confirmPassword: string;
  isAgreeTerms?: boolean;
  isReceiveNewsletters?: boolean;
}

export interface IActiveUser {
  uidb64: string | null;
  token: string | null;
}
