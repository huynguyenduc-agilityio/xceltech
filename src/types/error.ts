export type ErrorType = {
  detail: string;
};
