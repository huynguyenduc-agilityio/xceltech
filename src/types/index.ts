export * from './status';
export * from './employee';
export * from './table';
export * from './auth';
export * from './form';
export * from './tab';
export * from './user';
export * from './queryKey';
export * from './leave';
export * from './filter';
export * from './error';
export * from './notification';
export * from './profile';
