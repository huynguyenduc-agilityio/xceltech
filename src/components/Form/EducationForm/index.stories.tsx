import type { Meta, StoryObj } from '@storybook/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Components
import EducationForm from '.';

// Types
import { EducationType, MutationType } from '@/types';

const queryClient = new QueryClient();

const meta = {
  title: 'Components/EducationForm',
  component: EducationForm,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  render: (args) => (
    <QueryClientProvider client={queryClient}>
      <div className="flex items-center justify-center w-screen py-20 bg-blue-light">
        <div className="bg-white w-[988px] rounded-[15px]">
          <div className="p-10">
            <EducationForm {...args} />
          </div>
        </div>
      </div>
    </QueryClientProvider>
  ),
} satisfies Meta<typeof EducationForm>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    mode: MutationType.Create,
    section: EducationType.Academic,
  },
};
