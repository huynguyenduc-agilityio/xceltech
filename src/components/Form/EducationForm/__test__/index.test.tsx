import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { fireEvent, render, waitFor } from '@testing-library/react';

// Types
import { EducationType, MutationType } from '@/types';

// Components
import EducationForm from '..';

const renderComponent = () => {
  const queryClient = new QueryClient();

  return render(
    <QueryClientProvider client={queryClient}>
      <EducationForm
        mode={MutationType.Create}
        section={EducationType.Academic}
      />
      ,
    </QueryClientProvider>,
  );
};

describe('EducationForm Component', () => {
  it('should match snapshot', () => {
    const container = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('handle input value change and form submit', () => {
    const { getByPlaceholderText, getByRole } = renderComponent();
    const nameInput = getByPlaceholderText('Enter your institution name');
    const courseInput = getByPlaceholderText('Enter your course');
    const departmentInput = getByPlaceholderText('Enter your department');
    const locationInput = getByPlaceholderText('Enter your location');
    const submitButton = getByRole('button', { name: /submit/i });

    fireEvent.change(nameInput, { target: { value: 'Test Name' } });
    fireEvent.change(courseInput, { target: { value: 'Test Course' } });
    fireEvent.change(departmentInput, { target: { value: 'Test Department' } });
    fireEvent.change(locationInput, { target: { value: 'Test Location' } });

    fireEvent.click(submitButton);

    expect(nameInput).toHaveValue('Test Name');
    expect(courseInput).toHaveValue('Test Course');
    expect(departmentInput).toHaveValue('Test Department');
    expect(locationInput).toHaveValue('Test Location');
  });

  it('should show error message fields', async () => {
    const { getByPlaceholderText, getByText } = renderComponent();
    const nameInput = getByPlaceholderText('Enter your institution name');
    const courseInput = getByPlaceholderText('Enter your course');
    const departmentInput = getByPlaceholderText('Enter your department');
    const locationInput = getByPlaceholderText('Enter your location');

    fireEvent.change(nameInput, { target: { value: '' } });
    fireEvent.change(courseInput, { target: { value: '' } });
    fireEvent.change(departmentInput, { target: { value: '' } });
    fireEvent.change(locationInput, { target: { value: '' } });

    fireEvent.blur(nameInput);
    fireEvent.blur(courseInput);
    fireEvent.blur(departmentInput);
    fireEvent.blur(locationInput);

    await waitFor(() => {
      expect(getByText('Name of Institution is required!')).toBeInTheDocument();
      expect(getByText('Course is required!')).toBeInTheDocument();
      expect(getByText('Department is required!')).toBeInTheDocument();
      expect(getByText('Location is required!')).toBeInTheDocument();
    });
  });
});
