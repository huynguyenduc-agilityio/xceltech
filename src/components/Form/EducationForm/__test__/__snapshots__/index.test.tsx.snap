// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationForm Component should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="w-full text-black-default space-y-10"
        >
          <div
            class="space-y-8 mt-12"
          >
            <div
              class="flex items-start gap-9"
            >
              <div
                class="grid w-full gap-2"
              >
                <div
                  class="w-full space-y-5"
                >
                  <label
                    class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                    id="Name of Institution-label"
                  >
                    Name of Institution
                  </label>
                  <input
                    aria-describedby=":r0:-form-item-description"
                    aria-invalid="false"
                    aria-labelledby="Name of Institution-label"
                    autocomplete="off"
                    class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                    data-testid="input-element"
                    id=":r0:-form-item"
                    name="name"
                    placeholder="Enter your institution name"
                    value=""
                  />
                </div>
              </div>
              <div
                class="grid w-full gap-2"
              >
                <div
                  class="w-full space-y-5"
                >
                  <label
                    class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                    id="Department-label"
                  >
                    Department
                  </label>
                  <input
                    aria-describedby=":r1:-form-item-description"
                    aria-invalid="false"
                    aria-labelledby="Department-label"
                    autocomplete="off"
                    class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                    data-testid="input-element"
                    id=":r1:-form-item"
                    name="department"
                    placeholder="Enter your department"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="flex items-start gap-9"
            >
              <div
                class="grid w-full gap-2"
              >
                <div
                  class="w-full space-y-5"
                >
                  <label
                    class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                    id="Course-label"
                  >
                    Course
                  </label>
                  <input
                    aria-describedby=":r2:-form-item-description"
                    aria-invalid="false"
                    aria-labelledby="Course-label"
                    autocomplete="off"
                    class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                    data-testid="input-element"
                    id=":r2:-form-item"
                    name="course"
                    placeholder="Enter your course"
                    value=""
                  />
                </div>
              </div>
              <div
                class="grid w-full gap-2"
              >
                <div
                  class="w-full space-y-5"
                >
                  <label
                    class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                    id="Location-label"
                  >
                    Location
                  </label>
                  <input
                    aria-describedby=":r3:-form-item-description"
                    aria-invalid="false"
                    aria-labelledby="Location-label"
                    autocomplete="off"
                    class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                    data-testid="input-element"
                    id=":r3:-form-item"
                    name="location"
                    placeholder="Enter your location"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="flex items-start gap-9"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft flex flex-1 text-md"
                >
                  Start Date
                </label>
                <button
                  aria-controls="radix-:r5:"
                  aria-expanded="false"
                  aria-haspopup="dialog"
                  aria-label="pick-a-date"
                  class="font-productsans inline-flex items-center whitespace-nowrap rounded-[14px] ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed hover:bg-accent hover:text-accent-foreground w-full h-[54px] justify-between py-4 text-lg rounded-regular font-normal bg-blue-light text-black-smoky px-11"
                  data-state="closed"
                  type="button"
                >
                  <span
                    class="text-gray-soft"
                  >
                    Enter your start date
                  </span>
                  <svg
                    class="w-[23px] h-[27px]"
                    fill="none"
                    height="27px"
                    style="color: rgb(156, 156, 156);"
                    viewBox="0 0 23 27"
                    width="23px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 24.469C0 25.866 1.104 27 2.464 27h18.072C21.896 27 23 25.866 23 24.469V10.125H0zm16.429-10.336c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm-6.572-6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm-6.571-6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm17.25-17.508H18.07V.844A.835.835 0 0 0 17.25 0h-1.643a.835.835 0 0 0-.821.844v2.531H8.214V.844A.835.835 0 0 0 7.393 0H5.75a.835.835 0 0 0-.821.844v2.531H2.464C1.104 3.375 0 4.509 0 5.906v2.532h23V5.906c0-1.397-1.104-2.531-2.464-2.531"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft flex flex-1 text-md"
                >
                  End Date
                </label>
                <button
                  aria-controls="radix-:r7:"
                  aria-expanded="false"
                  aria-haspopup="dialog"
                  aria-label="pick-a-date"
                  class="font-productsans inline-flex items-center whitespace-nowrap rounded-[14px] ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed hover:bg-accent hover:text-accent-foreground w-full h-[54px] justify-between py-4 text-lg rounded-regular font-normal bg-blue-light text-black-smoky px-11"
                  data-state="closed"
                  type="button"
                >
                  <span
                    class="text-gray-soft"
                  >
                    Enter your end date
                  </span>
                  <svg
                    class="w-[23px] h-[27px]"
                    fill="none"
                    height="27px"
                    style="color: rgb(156, 156, 156);"
                    viewBox="0 0 23 27"
                    width="23px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 24.469C0 25.866 1.104 27 2.464 27h18.072C21.896 27 23 25.866 23 24.469V10.125H0zm16.429-10.336c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm-6.572-6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm-6.571-6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm17.25-17.508H18.07V.844A.835.835 0 0 0 17.25 0h-1.643a.835.835 0 0 0-.821.844v2.531H8.214V.844A.835.835 0 0 0 7.393 0H5.75a.835.835 0 0 0-.821.844v2.531H2.464C1.104 3.375 0 4.509 0 5.906v2.532h23V5.906c0-1.397-1.104-2.531-2.464-2.531"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="w-full space-y-5"
          >
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
            >
              Description
            </label>
            <textarea
              aria-describedby=":r8:-form-item-description"
              aria-invalid="false"
              class="flex min-h-[80px] w-full rounded-regular bg-blue-light py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none resize-none disabled:cursor-not-allowed disabled:opacity-50 h-[122px] px-9 rounded-[15px]"
              id=":r8:-form-item"
              name="description"
              placeholder="Enter your description"
            />
          </div>
          <div
            class="w-full h-[70px]"
          >
            <button
              class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white h-[70px] py-6 w-[364px] bg-green-primary"
              disabled=""
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
      ,
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="w-full text-black-default space-y-10"
      >
        <div
          class="space-y-8 mt-12"
        >
          <div
            class="flex items-start gap-9"
          >
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Name of Institution-label"
                >
                  Name of Institution
                </label>
                <input
                  aria-describedby=":r0:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Name of Institution-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r0:-form-item"
                  name="name"
                  placeholder="Enter your institution name"
                  value=""
                />
              </div>
            </div>
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Department-label"
                >
                  Department
                </label>
                <input
                  aria-describedby=":r1:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Department-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r1:-form-item"
                  name="department"
                  placeholder="Enter your department"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="flex items-start gap-9"
          >
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Course-label"
                >
                  Course
                </label>
                <input
                  aria-describedby=":r2:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Course-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r2:-form-item"
                  name="course"
                  placeholder="Enter your course"
                  value=""
                />
              </div>
            </div>
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Location-label"
                >
                  Location
                </label>
                <input
                  aria-describedby=":r3:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Location-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r3:-form-item"
                  name="location"
                  placeholder="Enter your location"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="flex items-start gap-9"
          >
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft flex flex-1 text-md"
              >
                Start Date
              </label>
              <button
                aria-controls="radix-:r5:"
                aria-expanded="false"
                aria-haspopup="dialog"
                aria-label="pick-a-date"
                class="font-productsans inline-flex items-center whitespace-nowrap rounded-[14px] ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed hover:bg-accent hover:text-accent-foreground w-full h-[54px] justify-between py-4 text-lg rounded-regular font-normal bg-blue-light text-black-smoky px-11"
                data-state="closed"
                type="button"
              >
                <span
                  class="text-gray-soft"
                >
                  Enter your start date
                </span>
                <svg
                  class="w-[23px] h-[27px]"
                  fill="none"
                  height="27px"
                  style="color: rgb(156, 156, 156);"
                  viewBox="0 0 23 27"
                  width="23px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 24.469C0 25.866 1.104 27 2.464 27h18.072C21.896 27 23 25.866 23 24.469V10.125H0zm16.429-10.336c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm-6.572-6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm-6.571-6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm17.25-17.508H18.07V.844A.835.835 0 0 0 17.25 0h-1.643a.835.835 0 0 0-.821.844v2.531H8.214V.844A.835.835 0 0 0 7.393 0H5.75a.835.835 0 0 0-.821.844v2.531H2.464C1.104 3.375 0 4.509 0 5.906v2.532h23V5.906c0-1.397-1.104-2.531-2.464-2.531"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft flex flex-1 text-md"
              >
                End Date
              </label>
              <button
                aria-controls="radix-:r7:"
                aria-expanded="false"
                aria-haspopup="dialog"
                aria-label="pick-a-date"
                class="font-productsans inline-flex items-center whitespace-nowrap rounded-[14px] ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed hover:bg-accent hover:text-accent-foreground w-full h-[54px] justify-between py-4 text-lg rounded-regular font-normal bg-blue-light text-black-smoky px-11"
                data-state="closed"
                type="button"
              >
                <span
                  class="text-gray-soft"
                >
                  Enter your end date
                </span>
                <svg
                  class="w-[23px] h-[27px]"
                  fill="none"
                  height="27px"
                  style="color: rgb(156, 156, 156);"
                  viewBox="0 0 23 27"
                  width="23px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 24.469C0 25.866 1.104 27 2.464 27h18.072C21.896 27 23 25.866 23 24.469V10.125H0zm16.429-10.336c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm-6.572-6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm-6.571-6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm17.25-17.508H18.07V.844A.835.835 0 0 0 17.25 0h-1.643a.835.835 0 0 0-.821.844v2.531H8.214V.844A.835.835 0 0 0 7.393 0H5.75a.835.835 0 0 0-.821.844v2.531H2.464C1.104 3.375 0 4.509 0 5.906v2.532h23V5.906c0-1.397-1.104-2.531-2.464-2.531"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="w-full space-y-5"
        >
          <label
            class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
          >
            Description
          </label>
          <textarea
            aria-describedby=":r8:-form-item-description"
            aria-invalid="false"
            class="flex min-h-[80px] w-full rounded-regular bg-blue-light py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none resize-none disabled:cursor-not-allowed disabled:opacity-50 h-[122px] px-9 rounded-[15px]"
            id=":r8:-form-item"
            name="description"
            placeholder="Enter your description"
          />
        </div>
        <div
          class="w-full h-[70px]"
        >
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white h-[70px] py-6 w-[364px] bg-green-primary"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
